bson/src/main/scala/DefaultBSONHandlers.scala

```diff
@@ -184,9 +166,6 @@ trait DefaultBSONHandlers {
   implicit def findWriter[T](implicit writer: VariantBSONWriter[T, _ <: BSONValue]): BSONWriter[T, _ <: BSONValue] =
     new VariantBSONWriterWrapper(writer)
 
-  implicit def findReader[T](implicit reader: VariantBSONReader[_ <: BSONValue, T]): BSONReader[_ <: BSONValue, T] =
```

Comment:
please note that I also remove this implicit, because I think it is never used...